birdie/src/lib/components/files/TreeNode.svelte

```diff
@@ -0,0 +1,106 @@
+<script lang="ts">
+	import { onDestroy, onMount } from 'svelte';
+	import { Button, TableBodyCell, TableBodyRow } from 'flowbite-svelte';
+	import {
+		FileCheckOutline,
+		FileCheckSolid,
+		FolderSolid,
+		HeartOutline,
+		HeartSolid
+	} from 'flowbite-svelte-icons';
+	import { FileType, LocalFileLFSState, type Node } from '$lib/types';
+	import { getFiles } from '$lib/repo';
+	import { fetchIncludeList, selectedFile } from '$lib/stores';
+
+	export let fileNode: Node;
+	export let loading: boolean;
+	export let level: number;
+
+	const getChildren = async () => {
+		if (fileNode.value.fileType === FileType.File) {
+			return;
+		}
+		fileNode.children = [];
+		const childFiles = await getFiles(fileNode.value.path);
+		const newChildren = childFiles.map(
+			(child) =>
+				({
+					value: child,
+					open: false,
+					children: []
+				} as Node)
+		);
+
+		fileNode = {
+			...fileNode, // spread the existing fileNode
+			children: newChildren
+		};
+	};
+
+	const handleOnClick = () => {
+		loading = true;
+		fileNode = {
+			...fileNode,
+			open: !fileNode.open
+		};
+		if (fileNode.open) {
+			void getChildren();
```

Comment:
curious should this getChildren call be awaited here so this handler doesn't accidentally set `loading = false` before the getChildren call finishes up?

Reply:
good catch, yea this whole function should be async